kroxylicious-kms-provider-kroxylicious-inmemory-test-support/src/main/java/io/kroxylicious/kms/provider/kroxylicious/inmemory/InMemoryTestKekManager.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kms.provider.kroxylicious.inmemory;
+
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.CompletionException;
+
+import io.kroxylicious.kms.service.TestKekManager;
+import io.kroxylicious.kms.service.UnknownAliasException;
+
+public class InMemoryTestKekManager implements TestKekManager {
+    private final InMemoryKms kms;
+
+    public InMemoryTestKekManager(InMemoryKms kms) {
+        this.kms = kms;
+    }
+
+    @Override
+    public UUID read(String alias) {
+        return kms.resolveAlias(alias).toCompletableFuture().join();
+    }
+
+    @Override
+    public void generateKek(String alias) {
+        Objects.requireNonNull(alias);
+
+        try {
+            read(alias);
+            throw new AlreadyExistsException(alias);
```

Comment:
I'd really like a 
```suggestion
            var existing = read(alias);
            if (existing != null) {
                throw new AlreadyExistsException(alias);
           }
```
I know its effectively the same but feels much less surprising to read.